src/include/sampleheap.hpp

```diff
@@ -136,13 +135,10 @@ class SampleHeap : public SuperHeap {
     SuperHeap::free(ptr);
     if (buf) {
       if (sz < buf_size) {
-        if (buf_size - sz <= PYMALLOC_MAX_SIZE) {
```

Comment:
This was an oversight that I made in previous commits, this line does not belong here